### PR TITLE
Bugfix/null element

### PR DESCRIPTION
**Pull request by DSergiu**
Closes https://github.com/hCaptcha/react-hcaptcha/issues

* Removes auto generated id, but will set if passed
* Replaces `document.getElementById` with a [React Ref](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs)
* Adds tests for `id` setting
* Bumps version to `0.3.1`
* Merged PR https://github.com/hCaptcha/react-hcaptcha/pull/62
* Removes redundant build files - cumbersome to keep up to date for every change

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -208,8 +193,9 @@ class HCaptcha extends React.Component {
     }
 
     render () {
+      // Keep elementId for backwards compatibility
       const { elementId } = this.state;
-      return <div id={elementId}></div>
+      return <div ref={this.myRef} id={elementId}></div>
```
Comment (brdlyptrs) [inline src/index.js:198]: Could we remove `elementId` then? Doesn't seem like its needed anymore.

File: src/index.js [hunk 2]
```diff
@@ -67,20 +50,22 @@ class HCaptcha extends React.Component {
       if (!isApiReady)
         captchaScriptCreated = false;
 
+      this.myRef = React.createRef();
```
Comment (brdlyptrs) [inline src/index.js:53]: Could you call this `this.ref`?

Comment (DSergiu) [reply src/index.js:53]: Sure thing.

File: src/index.js (hunk 1 above)
Comment (DSergiu) [reply src/index.js:198]: I chose to keep it to be backwards compatible.

**Review by brdlyptrs — DISMISSED** [review]

File: src/index.js [hunk 3]
```diff
@@ -34,6 +34,10 @@ class HCaptcha extends React.Component {
       super(props);
       const { id=null } = props;
 
+      if (id && console.warn) {
```
Comment (brdlyptrs) [inline src/index.js:37]: We want this to only be flagged in DEV mode, will this get stripped out?

**Review by mtratsiuk — APPROVED** [review]

**Pull request merged**